src/address_finder.rs

```diff
@@ -118,6 +123,50 @@ mod os_impl {
         }
     }
 
+    fn switch_ns(fd: RawFd) -> Result<(), Error> {
```

Comment:
to fix the fd lifetime issue, take it by reference. to be extra careful,

```rust
fn switch_ns<F: AsRawFd>(fd: &F) -> Result<(), Error> {
    let fd = fd.as_raw_fd();
    // ...
}
```

this means you need to have a thing that "is" a file descriptor,and that it has to outlive the call to `switch_ns`. you'd pass in the file instead of the fd below.